worker-document/src/main/java/com/hpe/caf/worker/document/scripting/specs/AbstractScriptSpec.java

```diff
@@ -51,14 +53,21 @@ public static AbstractScriptSpec create(final ApplicationImpl application, final
             throw new InvalidScriptException(script, "Script must have a single source.");
         }
 
+        final ScriptEngineType engineType;
+        if (script.engine == null || script.engine.isEmpty()) {
```

Comment:
Why make empty string valid?

```suggestion
        if (script.engine == null) {
```